hir/src/dialects/builtin/ops/component/interface.rs

```diff
@@ -69,18 +72,25 @@ impl TryFrom<&SymbolPath> for ComponentId {
         let mut components = path.components().peekable();
         components.next_if_eq(&SymbolNameComponent::Root);
 
-        let (ns, name) = match components.next().map(|c| c.as_symbol_name()) {
+        let (ns, name, version) = match components.next().map(|c| c.as_symbol_name()) {
             None => return Err(InvalidComponentIdError::MissingNamespace),
             Some(name) => match name.as_str().split_once(':') {
-                Some((ns, name)) => (SymbolName::intern(ns), SymbolName::intern(name)),
+                Some((ns, name)) => match name.split_once('@') {
+                    Some((name, version)) => (
+                        SymbolName::intern(ns),
+                        SymbolName::intern(name),
+                        Version::parse(version).map_err(InvalidComponentIdError::InvalidVersion)?,
+                    ),
+                    None => return Err(InvalidComponentIdError::MissingVersion),
```

Comment:
Is this truly an error? My recollection of the Component Model spec is that the version is supposed to be optional AFAIK

Reply:
Good catch! I've been staring at versions for so long, I forgot that it's optional. 
Should we make `ComponentId::version` optional? I don't like the idea of putting a hardcoded default version if it's not specified.

Reply:
I'll check and see what the spec indicates an elided version implies.

Reply:
According to the Component Model spec, the version is definitely optional, but it does not specify what a missing version implies. My guess is that it just means any version will satisfy it. As for what we should do with it, I'm not sure. Do you know if we always observe versioned paths in the compiled Wasm binary? If so, maybe we just require a version as you've sketched out here. Not sure if there is a good reason to allow paths with an unspecified version. I'll punt on this until we have a specific issue come up.

Reply:
So far, we have __only__ versioned paths in the compiled binaries. AFAIR, I could not use unversioned paths in WIT. 
I'd require a version for now, but since we're going to use digests for our dependencies, versions seem redundant, so I'd explore if there is a possibility to ditch them.